### PR TITLE
Warn when Cypress detects policies that may interfere with automating Chrome

### LINKED ISSUE #4391: Warn when ProxyMode/ExtensionWhiteList registry setting is detected
Cypress fails in strange ways if a ProxyMode policy is set on the user's computer, because it can break Cypress's proxy when running tests in Chrome.

We should:

* warn the user on the command line when Cypress is launched if we detect this setting
* maybe also warn the user if they try to launch Chrome with this setting enabled - could have chrome greyed out, or add a small :warning: sign or something with a tooltip explaining this problem - also don't set Chrome as the default browser

Relevant issues: #1253 and #4263

----

Should also detect ExtensionWhiteList/other problematic extension settings. 

Relevant issues: #1239, #2494

### PR DESCRIPTION
**Pull request by flotwig**
Fixes #4391

- [x] Add warnings in run mode
- [x] Add tooltip warning in open mode
- [x] Add link to tooltip
	- cypress-io/react-tooltip#4
- [x] Add support for multiple warnings in desktop-gui, since now there can be 2
	- Removed the second warning, but still keeping this code in for future use
- [x] Write documentation with workarounds. Documentation PR: cypress-io/cypress-documentation#1754 On-link PR: cypress-io/cypress-services#741

### Warning in run mode (only if they're using a Chrome-family browser):

```
Cypress detected policy settings on your computer that may cause issues. 

The following policies were detected that may prevent Cypress from automating Chrome: 

 > HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome\ProxyMode 

For more information, see https://on.cypress.io/bad-browser-policy
```

### Tooltip warning

image

### CODE REVIEW

**Comment by flotwig** [comment]
@jennifer-shehane 	I've updated it to https://on.cypress.io/bad-browser-policy , just not in the screenshots.

File: packages/server/lib/util/chrome_policy_check.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+module.exports = {
+  run: _.noop,
+  getRunner,
+}
+
+/**
+ * Only check on Windows. While it is possible for macOS/Linux to have preferences set that
+ * override Cypress's settings, it's never been reported as an issue and would require more
+ * native extensions to support checking.
```
Comment (jennifer-shehane) [inline packages/server/lib/util/chrome_policy_check.js]: maybe link to issue here

File: packages/server/lib/util/chrome_policy_check.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  'ExtensionInstallSources',
+  'ExtensionAllowedTypes',
+  'ExtensionAllowInsecureUpdates',
+  'ExtensionSettings',
+  'UninstallBlacklistedExtensions',
+]
+
+const POLICY_KEYS = [
+  'Software\\Policies\\Google\\Chrome',
+  'Software\\Policies\\Google\\Chromium',
```
Comment (jennifer-shehane) [inline packages/server/lib/util/chrome_policy_check.js]: This also works for Canary?

**Comment by wojciechsabaj** [comment]
Hi, great job!

In my opinion, the balloon tip will be enough when you move the content from yellow warning message to the balloon tip.

If you insist on leaving the yellow warning message, please make it more condensed, as the version presented on screenshots contains unnecessary newlines between sentences.

Keep working on Cypress! It's awesome, even with its limitations.

File: packages/desktop-gui/cypress/integration/project_nav_spec.coffee [hunk 3]
```diff
@@ -232,6 +232,32 @@ describe "Project Nav", ->
...
+          "warning": "Cypress detected policy settings on your computer that may cause issues with using this browser. For more information, see https://on.cypress.io/bad-browser-policy"
+        }]
+
+        @config.browsers = @browsers
+        @openProject.resolve(@config)
+
+      it "shows warning icon with linkified tooltip", ->
+        cy.get(".browsers .fa-exclamation-triangle")
+          .then ($el) ->
+            $el[0].dispatchEvent(new Event("mouseover", {bubbles: true}))
```
Comment (brian-mann) [inline packages/desktop-gui/cypress/integration/project_nav_spec.coffee]: should just be using `cy.trigger()` here...

`cy.get("...").trigger("mouseover")`

File: packages/server/package.json [hunk 4]
```diff
@@ -191,9 +191,6 @@
     "url-parse": "1.4.7",
     "uuid": "3.3.2",
     "widest-line": "3.1.0",
-    "winston": "1.1.2"
-  },
-  "optionalDependencies": {
-    "@cypress/registry-js": "1.3.3-patch1"
```
Comment (flotwig) [inline packages/server/package.json]: :thinking: I was wondering why this PR didn't seem to be having any effect!

**Pull request merged**